custom_components/frigate/media_source.py

```diff
@@ -437,13 +437,35 @@ def from_str(
         if len(parts) < 2 or parts[1] != cls.get_identifier_type():
             return None
 
+        print(f"The parts are {parts}")
+        tz = "UTC" if (len(parts) == 6 and str(parts[2]).count("-") == 2) else "LOCAL"
+
+        # TODO probably a better way to do this using Frigate APIs with recordings info
         try:
+            if tz == "LOCAL":
+                return cls(
+                    frigate_instance_id=parts[0],
+                    year_month=cls._get_index(parts, 2),
+                    day=cls._get_index(parts, 3),
+                    hour=cls._get_index(parts, 4),
+                    camera=cls._get_index(parts, 5),
+                )
+
+            ymd_parts = str(cls._get_index(parts, 2)).split("-")
+            hour = cls._get_index(parts, 3)
+            time = dt.datetime(
+                int(ymd_parts[0]),
+                int(ymd_parts[1]),
+                int(ymd_parts[2]),
+                int(hour),
+                tzinfo=dt.timezone.utc,
+            ).astimezone()
```

Comment:
Maybe there's a way to combine the validation above with the time construction here? i.e. `datetime.datetime.strptime(f"{parts[2]} {cls._get_index(parts, 3)}", "%Y-%m-%d %H")`

Reply:
Unfortunately I don't believe strptime provides a good way to tell it what timezone the values are in so that conversion would be more complicated

Reply:
I did try to make it a bit cleaner though, let me know what you think